K8s Support

akutz opened the pull request:
This patch adds support for CSI's centralized model (ex. K8s) by performing Controller Attach/Detach logic inside of the Node service's Pub/Unpub RPCs.

codenrhoden commented on service/node.go:
```diff
@@ -58,3 +311,30 @@ func (s *service) NodeGetCapabilities(
 	}, nil
 }
+
+func validateDevice(devicePath string) (string, error) {
```
`gofsutil.ValidateDevice` is not usable here?

akutz replied:
Hi @codenrhoden,

It didn't exist when I wrote the above code. I added `ValidateDevice` to `gofsutil` later. I will update to the latest version of gofsutil and use its `ValidateDevice`. Good catch.

codenrhoden commented on service/node.go:
```diff
@@ -1,25 +1,278 @@
+	}
+
+	// Attach the VMDK to this node.
+	dev, err := s.ops.Attach(req.VolumeId, nil)
```
I'm not familiar with how `vmdkops` works here, but will it gracefully handle if the device is already attached? I'm wondering if there is any code that can be short-circuited if the device is already attached. Thought maybe that isn't necessary, as that is left to the idempotent provider.

akutz replied:
Hi @codenrhoden,

The backend ESXi component's `attach` call is idempotent.

codenrhoden reviewed (approved):
Looks like this will take care of doing all the controller actions for attach/detach in the node pub/unpub requests instead. Looks good!

akutz committed:
Travis-CI Build Support

This patch includes support for Travis-CI.

akutz committed:
K8s Support

This patch adds support for CSI's centralized model (ex. K8s) by
performing Controller Attach/Detach logic inside of the Node service's
Pub/Unpub RPCs.
